### PR TITLE
Round 2 preamble parsing

### PR DESCRIPTION
**Pull request by cmc333333**
We need to be able to address preamble sections by label (rather than, e.g. p0-p3-p2). Further, we need to make it possible to use non-textual paragraphs (e.g. images, tables, extracts). This PR accomplishes the former by making use of our existing `ParagraphProcessor`s and opens the door for the latter.

Several changesets, though all but the last are setup. Setup includes:
* Adding warnings for tags we weren't expecting
* Separates out `NOTE` processor to ignore it's header tags
* Allow a set of paragraphs to begin with MARKERLESS paragraphs without needing a new level of indentation

Then the rewrite:
* Preprocess all of the preamble paragraphs to create a nested XML structure (using header tags as dividers)
* Use a regex to pull out the marker from, e.g. "III. Header Text"
* Make use of a tag matcher/`ParagraphProcessor` in the preamble

This should resolve eregs/notice-and-comment#15

### CODE REVIEW

**Comment by cmc333333** [comment]
Investigating the FEC issue. It's close, but not quite right.

**Comment by cmc333333** [comment]
FEC issue is fixed -- the integration test failure is failing the "right" way. @anthonygarvan confirmed

File: regparser/tree/xml_parser/note_processor.py [hunk 1]
```diff
@@ -5,6 +5,7 @@
 
 
 class IgnoreNotesHeader(paragraph_processor.BaseMatcher):
+    """We don't want to include "Note:" and "Notes:" headers"""
```
Comment (tadhg-ohiggins) [inline regparser/tree/xml_parser/note_processor.py]: Is this too generic? Am I being paranoid about running into something like a header starting “Noted in our consideration of this proposal were the following…”? If I'm not being too paranoid, something like `return xml.tag == 'HD' and xml.text.lower().split(":", 1)[0] in ("note", "notes")` might be better.

Comment (cmc333333) [inline regparser/tree/xml_parser/note_processor.py]: Agreed, following up shortly.

**Pull request merged**